src/solc_0.8/asset/ERC1155ERC721.sol

```diff
@@ -54,34 +53,47 @@ contract ERC1155ERC721 is WithSuperOperators, IERC1155, IERC721 {
 
     mapping(address => address) private _creatorship; // creatorship transfer
 
+    // @review Bouncer
     mapping(address => bool) private _bouncers; // the contracts allowed to mint
-    mapping(address => bool) private _metaTransactionContracts; // native meta-transaction support
-
+    mapping(address => bool) private _metaTransactionProcessors; // meta-transaction support
+    // @review Bouncer ... Deprecate?
```

Comment:
Bouncer is the contracts that allow to mint Asset.
we could hardcode the admin and change minter via upgrade.

Reply:
@wighawag 
So, we still can't use immutable variables with oz-upgrades.
Hardcoding the bouncerAdmin makes sense.
Perhaps modifying the init function to accept a minter address/array of address (instead of the bouncerAdmin address) could work. As you noted, an upgrade would be needed to add/remove minters. 
To be honest, there's not a lot of savings to be had here in terms of bytecode... but maybe still with it as part of the Asset refactoring/simplification.

Reply:
We could probably remove:
- `event BouncerAdminChanged(address oldBouncerAdmin, address newBouncerAdmin);`
- `function getBouncerAdmin()`
- `function changeBouncerAdmin()`
- `function isBouncer()`
- `function setBouncer()`
and just keep the `Bouncer` event so that it's easy to track off-chain.

Reply:
I like isBouncer for our deploy script as fetching event is more cumbersome and slower